tools/azsdk-cli/Azure.Sdk.Tools.Cli.Tests/TestHelpers/TempDirectory.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Sdk.Tools.Cli.Helpers;
+
+namespace Azure.Sdk.Tools.Cli.Tests.TestHelpers;
+
+/// <summary>
+/// Utility to create and reliably clean up a temporary directory for tests.
+/// Implements both IDisposable and IAsyncDisposable so it can be used with
+/// either synchronous or asynchronous test flows. Deletion is retried to
+/// tolerate transient file locks on some platforms.
+/// </summary>
+public sealed class TempDirectory : IDisposable, IAsyncDisposable
+{
+    public string DirectoryPath { get; }
+    private bool _disposed;
+
+    private TempDirectory(string directoryPath)
+    {
+        DirectoryPath = directoryPath;
+        Directory.CreateDirectory(directoryPath);
+    }
+
+    /// <summary>
+    /// Creates a new temporary directory with an optional prefix.
+    /// </summary>
+    public static TempDirectory Create(string? prefix = null)
+    {
+        var name = (prefix ?? "azsdk-temp") + "_" + Guid.NewGuid().ToString("N");
+        var fullPath = RealPath.GetRealPath(Path.Combine(Path.GetTempPath(), name));
+        return new TempDirectory(fullPath);
+    }
+
+    public void Dispose()
+    {
+        if (_disposed)
+        {
+            return;
+        }
+
+        _disposed = true;
+        TryDeleteWithRetries(DirectoryPath);
+        GC.SuppressFinalize(this);
+    }
+
+    public ValueTask DisposeAsync()
+    {
+        Dispose();
+        return ValueTask.CompletedTask;
+    }
+
+    private static void TryDeleteWithRetries(string path, int attempts = 5, int initialDelayMs = 40)
+    {
+        if (!Directory.Exists(path))
+        {
+            return;
+        }
+
+        for (int i = 0; i < attempts; i++)
```

Comment:
I think it's better if we try to delete a directory once and then just move on if it fails. Directory names for tests should arguably be unique to prevent collision anyways, deletion is just a courtesy. Given that this could be used in the teardown of numerous tests I don't it's a good idea to be adding non-obvious sleeps across tests. Alternatively implement `DisposeAsync` so it actually does the retry in a non-blocking manner, but even that could have some issues.

Reply:
Or at least make the retry window much tighter, like 1ms. We don't need to be waiting 200ms per directory in the worst cases.